base/util.jl

```diff
@@ -576,4 +576,4 @@ a starting `crc` integer to be mixed in with the checksum.
 """
 function crc32c end
 crc32c(a::Array{UInt8}, crc::UInt32=0x00000000) = ccall(:jl_crc32c, UInt32, (UInt32, Ptr{UInt8}, Csize_t), crc, a, sizeof(a))
-crc32c(s::String, crc::UInt32=0x00000000) = crc32c(s.data, crc)
+crc32c(s::String, crc::UInt32=0x00000000) = ccall(:jl_crc32c, UInt32, (UInt32, Ptr{UInt8}, Csize_t), crc, s, sizeof(s))
```

Comment:
Maybe just use `Union{Array{UInt8},String}` for the first argument in a single method?